runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixOverlay.java

```diff
@@ -55,8 +56,19 @@ private TearsOfGuthixOverlay(TearsOfGuthixPlugin plugin)
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{

```

Comment:
Actually, one more optimisation I can think of would be to have `if (plugin.getStreams().size() == 0) { return null; }` here. It would avoid having to hit up the config and set up colors on every render pass when there aren't even any streams active.

Reply:
Not at all! I appreciate a finely tuned machine. I regret earlier comments that might indicate otherwise where I prioritized expediency over thoroughness. Split attention indeed..